custom_components/cololight/light.py

```diff
@@ -268,7 +267,18 @@ async def async_added_to_hass(self):
             self._effect = last_state.attributes.get("effect")
             self._brightness = last_state.attributes.get("brightness", 255)
             self._hs_color = last_state.attributes.get("hs_color")
-
+    
+    def update(self):
+        self._light._sock.sendto(bytes.fromhex("535a303000000000001e000000000000000000000000000000000200000000000000000003020101"), (self._host, self._port))
+        data = self._light._sock.recvfrom(4096)[0]
+        if not data: return
+        if data[40] == 207:#0xcf
+            self._on = True
+        elif data[40] == 206:#0xce
+           self._on = False
+        else:
+            return #if value is not 0xcf(on) or 0xce(off) stop the update then dont change anything
```

Comment:
Out of curiosity, what state can be returned other than on and off?

Reply:
The only states that the lamp returns are on or off. The official cololight app does not support receiving the effect(if you change the effect and reopen the app, it will default to "savasana" effect). But in my testing sometimes the light will return something else.(I think if it receives a on or off command). Thats why I made that else statement.

Reply:
I have been doing some testing, and have found that the response can return either state/brightness or effect/colour (this is why `data[40]` sometimes returns something other than state (on/off)).

However I haven't been able to get it to consistently return either state/brightness or effect/colour.
Sometimes by calling twice, you can get both (whereby the first call returns  state/brightness, and then the second call returns effect/colour).